### ISSUE TITLE
Import creates extra property groups not present in the definition

### ISSUE THREAD

**Issue by bachratyg** [KevinJump/uSync8#295]
**Describe the bug**
Using Umbraco 8.17.0 and uSync 8.10.3 the content type import is broken. Import creates extra property groups if the property name is significantly different from the alias

**To Reproduce**
1. Create a new content type
2. Add several properties, edit the alias manually to be very different from the name (e.g. german name and english alias)
3. Init an empty database, do a clean install and the initial import
5. Open the page of the imported content type. There are property groups those name matches imported properties. Some may contain properties, some are empty.

**Expected behavior**
No extra items that are not present in the def config, i.e. same behavior as with uSync 8.9.7 on Umbraco 8.17.0 when there are no tabs in the definitions.

**About your Site (please complete the following information):**
 - Umbraco Version: 8.17.0
 - uSync Version: 8.10.3
 - Browser N/A

**Additional context**
I believe the problem is here:
https://github.com/KevinJump/uSync/blob/7265e452c908ef16742b9d0e20dd7dd9fb7dff4d/uSync8.Core/Serialization/Serializers/ContentTypeBaseSerializer.cs#L659
and here
https://github.com/KevinJump/uSync/blob/7265e452c908ef16742b9d0e20dd7dd9fb7dff4d/uSync8.Core/Serialization/Serializers/ContentTypeBaseSerializer.cs#L467

The API expects the alias, but the name gets passed in. This is fine as long as the alias is automatically generated from the name (the lookup tries to reconstruct the alias if an exact match is not found) but fails to find the correct item if the alias and the name differ significantly. It looks like v9 already handles aliases correctly e.g.: https://github.com/KevinJump/uSync/blob/6549b1ce158b2240e42b00a2d98e4166d501944e/uSync.Core/Serialization/Serializers/ContentTypeBaseSerializer.cs#L739

**Comment by bachratyg** [comment]
This may also be related:
https://github.com/KevinJump/uSync/blob/7265e452c908ef16742b9d0e20dd7dd9fb7dff4d/uSync8.Core/Extensions/PropertyGroupExtensions.cs#L117
`AddPropertyType(PropertyType, string, string)` is declared on the base interface (`IContentTypeBase`) thus `addPropertyTypeMethod` will always be null and only the fallback code runs.

**Comment by KevinJump** [comment]
yeah 😞  - i expect i got confused, the internal Umbraco api has changed to support tabs and use to take name and now takes alias, so looks like i missed it in a few places.

**Comment by KevinJump** [comment]
yeah some of those methods are the pre v8.17 methods so the extension call, is finding the v8.17 version via reflection and calling that. if the version is pre 8.17 then the old legacy method is called. 

i think the cleans are wrong, i need to check the others _(pain to test as the core project is running 8.1 and the methods don't actually exist)._

**Comment by KevinJump** [comment]
i think this does it - I've renamed a few things because what is the name and what is the alias faling back to the name gets confusing. 

I think changing the signatures worked 👍 .

we have a nightly build for testing : https://dev.azure.com/jumoo/Public/_packaging?_a=package&feed=nightly&package=uSync&protocolType=NuGet&version=8.10.4-preview.20211013.3&view=overview


Just to confirm when you say : 

> 2. Add several properties, edit the alias manually to be very different from the name (e.g. german name and english alias)

do you mean:
-  edit the property alias (i have tried this) 
- or go into the uSync file and edit the `Caption` of the tab (i've also done that) 

just want to understand what was breaking for you.

**Comment by bachratyg** [comment]
Our usual workflow is having a dev db (per dev), everyone configures required data types/content types/etc in the backoffice, automatically written to disk by uSync, then committed with the rest of the code. No direct uSync file editing, except for some very special cases e.g. merge conflicts. The properties and descriptions are in the customer's language, but the aliases are always english.

We only ever publish the compiled app and configuration, the database is always handled by migrations and uSync. What the users are allowed to modify is strictly separated from what devs are allowed to publish to prevent conflicts.

The deployment package is required to be able to
1. perform an update of a previously deployed version with no developer intervention and no manual steps from the customer.
This is tested quite often and without much effort: when a dev pushes new stuff it should just work immediately for other devs.
2. perform a clean install on an empty database
This is usually tested after a dependency upgrade or some major new feature.

Deps are upgraded in a single wave: bump all, test once. When uSync is upgraded we do an export to ensure that all files have the same format. Some minor changes are expected due to dev error (e.g. the content type references a nonexistent template), this time however pretty much every file had some additional property groups and curiously they had the same name as existing properties.

To rule out the possibility that the Umbraco upgrade (8.15 to 8.17) screwed up existing content types through some migration I reverted all changes and tried a clean install and clean export. This is also useful to test that definitions roundtrip properly: nothing gets added, nothing is left out, defs are a solid source of truth. Ideally this only removes legacy (delete/remove) entries from the files, but alas the same extra property groups appeared.

I'll test the nightly as soon as I'm able to, but it might take a couple of days.

**Comment by KevinJump** [comment]
Hi, we've pushed this as a new release (v8.10.4) because it turns out we broke this on a bigger scale #297 .

I've checked the process and it seems to work now on both v8.16 and v8.17 i think it was actually a rouge swapping of the property name and the tab name in one of our add methods 🤦

**Comment by bachratyg** [comment]
Clean install seems to work now, the defs are mostly unchanged, there are some new nodes
* ContentType/MediaType/MemberType
  * new `Alias="..."` attribute on GenericProperties/GenericProperty/Tab
  * new `<Alias>...</Alias>` and `<Type>Group</Type>` element under Tabs/Tab
* DataType
  * `"OverlaySize": null` under config

However if upgrading an existing site all property groups and contained properties are lost. :(

What I noticed is that Umbraco and uSync generates a different alias from the name of the group. E.g. for name "Paraméterek" uSync uses the exact same expression, Umbraco uses "paramterek" - camel cased and non-ascii removed. Maybe a `ToSafeAlias(camel: true)` is still missing somewhere? Umbraco's migration code is here for reference:

https://github.com/umbraco/Umbraco-CMS/blob/213d8c00aa6655b2b5c1514c337221898b3c2ed9/src/Umbraco.Core/Migrations/Upgrade/V_8_17_0/AddPropertyTypeGroupColumns.cs#L33

**Comment by KevinJump** [comment]
The more i get into this, the more i realize it tabs update in Umbraco was a dogs dinner :(, we have methods that use to take name that now take alias. we can't just safe alias the name at the start, because we have to be able to find the tab by name incase its not been created by alias. 😞 

anyway - think its fix, - we will add this to the next release, (not going to push a release just for this, but there will be a nightly if you want to update to it).

**Comment by bachratyg** [comment]
No change with the nightly unfortunately. Still investigating what goes wrong...

Umm... is bca0874 already included in the nightly (8.10.4-preview.20211014.1) or should I wait for the next build? DevOps does not show the time zone of the publish date

**Comment by bachratyg** [comment]
There's still a name-alias mismatch in `CleanTabs` here:
https://github.com/KevinJump/uSync/blob/d6fc5602aea0be735a9afaa1fff392bb77981295/uSync8.Core/Serialization/Serializers/ContentTypeBaseSerializer.cs#L665
`newTabAliasOrNames` would contain the display names of tabs if the alias is not present in the def. However when building the `removals` it's compared to the alias of the tabs which was already created by the U17 migration. These naturally do not match.

Better to use the same lookup mechanism as in `DeserializeTabs`:
```
var newTabs = new List<PropertyGroup>();
foreach (var nameOrAlias in newTabAliasOrNames)
{
    var tab = item.PropertyGroups.FindTab(nameOrAlias);
    if( tab != null )
        newTabs.Add(tab);
}
var removals = newTabs.Except(item.PropertyGroups).Select(t => t.GetTabAliasOrName()).ToList();
```
